### ISSUE TITLE
New commit breaks linking in other builds (Ascent)

### ISSUE THREAD

**Issue by stevenwalton** [Alpine-DAV/vtk-h#46]
Building Ascent I am getting the error with VTK-h (relevant output from `make VERBOSE=1`)

```
[ 19%] Building CXX object ascent/CMakeFiles/ascent_mpi.dir/runtimes/ascent_vtkh_data_adapter.cpp.o
cd /home/ascent/Programming/src/ascent/build/ascent && /usr/bin/mpicxx  -DPARALLEL -DUSE_MPI -Dascent_mpi_EXPORTS -I/home/ascent/Programming/src/ascent/src/thirdparty_builtin/lodepng -I/home/ascent/Progra
mming/src/ascent/src/ascent -I/home/ascent/Programming/src/ascent/build/ascent -I/home/ascent/Programming/src/ascent/src/ascent/utils -I/home/ascent/Programming/src/ascent/src/ascent/runtimes -I/home/asce
nt/Programming/src/ascent/src/flow -I/home/ascent/Programming/src/ascent/build/flow -I/home/ascent/Programming/src/ascent/src/flow/filters -I/home/ascent/Programming/src/ascent/src/ascent/runtimes/flow_fi
lters -I/home/ascent/Programming/src/ascent/build/fortran -isystem /usr/lib/openmpi/include -isystem /usr/lib/openmpi/lib -I/home/ascent/Programming/Programs/conduit/include/conduit -I/home/ascent/Program
ming/Programs/vtk-h/include -isystem /usr/lib/openmpi/include/openmpi/opal/mca/event/libevent2021/libevent -isystem /usr/lib/openmpi/include/openmpi/opal/mca/event/libevent2021/libevent/include -isystem /
usr/lib/openmpi/include/openmpi -isystem /home/ascent/Programming/Programs/vtk-m/include/vtkm-1.2 -isystem /home/ascent/Programming/Programs/vtk-m/include/vtkm-1.2/vtkm/thirdparty/diy -isystem /home/ascen
t/Programming/Programs/vtk-m/include/vtkm-1.2/vtkm/thirdparty/taotuple  -g -fPIC   -D ASCENT_MPI_ENABLED  -std=c++11 -o CMakeFiles/ascent_mpi.dir/runtimes/ascent_vtkh_data_adapter.cpp.o -c /home/ascent/Pr
ogramming/src/ascent/src/ascent/runtimes/ascent_vtkh_data_adapter.cpp
/home/ascent/Programming/src/ascent/src/ascent/runtimes/ascent_vtkh_data_adapter.cpp: In static member function ‘static vtkh::DataSet* ascent::VTKHDataAdapter::BlueprintToVTKHDataSet(const conduit::Node&,
 const string&)’:
/home/ascent/Programming/src/ascent/src/ascent/runtimes/ascent_vtkh_data_adapter.cpp:128:68: error: ‘GetMPIComm’ is not a member of ‘vtkh’
     MPI_Allgather(&boolean, 1, MPI_INT, has_ids_array, 1, MPI_INT, vtkh::GetMPIComm());
                                                                    ^
/home/ascent/Programming/src/ascent/src/ascent/runtimes/ascent_vtkh_data_adapter.cpp:130:67: error: ‘GetMPIComm’ is not a member of ‘vtkh’
     MPI_Allgather(&boolean, 1, MPI_INT, no_ids_array, 1, MPI_INT, vtkh::GetMPIComm());
                                                                   ^
/home/ascent/Programming/src/ascent/src/ascent/runtimes/ascent_vtkh_data_adapter.cpp:163:75: error: ‘GetMPIComm’ is not a member of ‘vtkh’
     MPI_Allgather(&num_domains, 1, MPI_INT, domains_per_rank, 1, MPI_INT, vtkh::GetMPIComm());
                                                                           ^
ascent/CMakeFiles/ascent_mpi.dir/build.make:350: recipe for target 'ascent/CMakeFiles/ascent_mpi.dir/runtimes/ascent_vtkh_data_adapter.cpp.o' failed
make[2]: *** [ascent/CMakeFiles/ascent_mpi.dir/runtimes/ascent_vtkh_data_adapter.cpp.o] Error 1
make[2]: Leaving directory '/home/ascent/Programming/src/ascent/build'
CMakeFiles/Makefile2:1567: recipe for target 'ascent/CMakeFiles/ascent_mpi.dir/all' failed
make[1]: *** [ascent/CMakeFiles/ascent_mpi.dir/all] Error 2
make[1]: Leaving directory '/home/ascent/Programming/src/ascent/build'
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2

```

It appears to come down to improper linking. 

But if I use an older version of VTK-h (i.e. the copy [jameskress](https://github.com/jameskress/vtk-h/tree/develop) has) I can build with no problems.

**Comment by cyrush** [comment]
@stevenwalton 

Thanks for the report, vtk-h develop is out of sync with ascent develop. 

I changed the vtk-h develop api, thinking I would have a smooth transition to include in ascent, but I hit some strange issues with openmpi. 

Still working to get to the bottom of it, I hope to resolve in the next day or so.

**Comment by cyrush** [comment]
@stevenwalton 

Got to the root of the issue that was undermining the update to new vtk-h in ascent. 
I had several typos that created linking issues, those are resolved with #47.

I also merged a PR in ascent to use the newest vtk-h. So ascent develop and vtk-h devleop are back in sync. I am going to close this issue, if you have any more problems let me know.
